File: leetcode-solutions/1480-running-sum-1D-array.cs
public class Solution {
    
    private void Sum(int[] nums, int i) {
        
        if (i >= nums.Length) {

            return;
        }
        
        nums[i] += nums[i - 1];
        
        Sum(nums, i + 1);
    }

    public int[] RunningSum(int[] nums) {
        
        Sum(nums, 1);
        return nums;
    }
}

File: leetcode-solutions/1480-running-sum-1D-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var runningSum = function(nums) {
    
    function sum(nums, i) {

        if (i >= nums.length) {
            return;
        }

        nums[i] += nums[i - 1];

        sum(nums, i + 1);
    }
    
    sum(nums, 1);
    return nums;
};

// Time: O(n);
// Space: no additional space
